File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = 0;
        // go through each value in nums
        for (int i = 0; i < nums.size(); i++) {
            res ^= nums[i];
            res ^= i;
        }
        res ^= nums.size();

        return res;
    }
};

File: Data Structures & Algorithms/reverse-bits/submission-1.java
class Solution {
    public int reverseBits(int n) {
        int res = 0; //000000... 0

        for (int i = 0; i < 32; i++) {
            int val = (n >> i) & 1;
            res |= (val <<= 31 - i);
        }

        return res;
    }
}
